lib/src/simulations/epicycloid_curve.dart

```diff
@@ -0,0 +1,385 @@
+import 'dart:io';
+import 'dart:math';
+import 'dart:ui';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+GlobalKey<_EpicycloidState> globalKey = GlobalKey<_EpicycloidState>();
+
+class EpicycloidCurve extends StatefulWidget {
+  @override
+  _EpicycloidCurveState createState() => _EpicycloidCurveState();
+}
+
+class _EpicycloidCurveState extends State<EpicycloidCurve> {
+  double factor = 0;
+  double total = 0;
+  bool animatefactor = false;
+  bool animatepoints = false;
+  bool animating = false;
+
+  @override
+  void initState() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+    ]);
+    super.initState();
+  }
+
+  @override
+  dispose() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+      DeviceOrientation.landscapeLeft,
+      DeviceOrientation.landscapeRight,
+    ]);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    ScreenUtil.instance = ScreenUtil(
+      width: 512.0,
+      height: 1024.0,
+      allowFontScaling: true,
+    )..init(context);
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        leading: IconButton(
+          icon: Icon(Icons.arrow_back_ios),
+          onPressed: () {
+            Navigator.pop(context);
+          },
+        ),
+        title: Text(
+          'Epicycloid Pattern',
+          style: Theme.of(context).textTheme.title,
+        ),
+        centerTitle: true,
+      ),
+      floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
+      floatingActionButton: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Visibility(
+          visible: animatefactor || animatepoints,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: <Widget>[
+              FloatingActionButton(
+                  heroTag: 0,
```

Comment:
can you check that is this seriously needed, I don't get why this is causing an issue.

Reply:
I think the error arises because FAB being the primary button, there can only be 1 per screen by default. So, I guess as we aren't using hero animation here we can set them both to null.